src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -786,3 +801,90 @@ export const selectAccountIdByAccountNumberAndChainId = createSelector(
     })
   },
 )
+
+export const selectEquitiesFromFilter = createDeepEqualOutputSelector(
+  selectAccountIdsByAssetIdAboveBalanceThresholdByFilter,
+  selectPortfolioFiatBalancesByAccountId,
+  selectPortfolioAccountBalancesBaseUnit,
+  selectAllEarnUserLpOpportunitiesByFilter,
+  selectAllEarnUserStakingOpportunitiesByFilter,
+  selectAssets,
+  selectAssetIdParamFromFilter,
+  (
+    accountIds,
+    portfolioFiatBalances,
+    portfolioCryptoBalancesBaseUnit,
+    lpOpportunities,
+    stakingOpportunities,
+    assets,
+    assetId,
+  ): AssetEquity[] => {
+    if (!assetId) return []
+    const asset = assets[assetId]
+    const accounts = accountIds.map(accountId => {
+      const fiatAmount = bnOrZero(portfolioFiatBalances[accountId][assetId]).toString()
+      const cryptoAmountBaseUnit = bnOrZero(
+        portfolioCryptoBalancesBaseUnit[accountId][assetId],
+      ).toString()
+      return {
+        id: accountId,
+        type: AssetEquityType.Account,
+        fiatAmount,
+        provider: 'wallet',
+        cryptoAmountBaseUnit,
+        color: asset?.color,
+      }
+    })
+    const staking = stakingOpportunities.map(stakingOpportunity => {
+      // Because the underlying assets can have different precisions
+      // We need to convert it to the asset we are viewing to get correct amounts to sum together.
+      const underlyingAssetPrecision =
+        assets[stakingOpportunity.assetId]?.precision ?? asset?.precision
+      const cryptoAmountBaseUnit = convertPrecision({
+        value: bnOrZero(stakingOpportunity.cryptoAmountBaseUnit),
+        inputExponent: underlyingAssetPrecision,
+        outputExponent: asset?.precision ?? 0,
+      }).toString()
+      return {
+        id: stakingOpportunity.id,
+        type: AssetEquityType.Staking,
+        fiatAmount: stakingOpportunity.fiatAmount,
+        cryptoAmountBaseUnit,
+        underlyingAssetId: stakingOpportunity.underlyingAssetId,
+        provider: stakingOpportunity.provider,
+        color: DefiProviderMetadata[stakingOpportunity.provider].color,
+      }
+    })
+    const lp = lpOpportunities.map(lpOpportunity => {
+      return {
+        id: lpOpportunity.id,
+        type: AssetEquityType.LP,
+        fiatAmount: lpOpportunity.fiatAmount,
+        cryptoAmountBaseUnit: bnOrZero(lpOpportunity.cryptoAmountBaseUnit).toString(),
+        provider: lpOpportunity.provider,
+        color: DefiProviderMetadata[lpOpportunity.provider].color,
+      }
+    })
+    return [...accounts, ...lp, ...staking].sort((a, b) =>
```

Comment:
not a big deal but in general we should avoid spreading for perf where we can, this is very allocatey/GCy https://pouyae.medium.com/what-is-the-es6-spread-operator-and-why-you-shouldnt-use-it-57c056078ed9

Reply:
i.e. `accounts.concat(lp).concat(staking).sort` is much faster